deploy/helm/kubecf/templates/_stacks.tpl

```diff
@@ -0,0 +1,137 @@
+{{- /*
+==========================================================================================
+| _stacks.update $
++-----------------------------------------------------------------------------------------
+| For each $STACK:
+|
+| * Update the stemcell and url of each release from the "$defaults"
+|
+| * Sets stacks.$STACK.enabled depending on install_stacks
+| * Sets stacks.$STACK.install_buildpacks to [prepend, list, append]
+| * Sets stacks.$STACK.buildpack to map buildpack short names to releases
+|
+| For each $RELEASE (in each $STACK):
+|
+| * Sets stacks.$STACK.releases.$RELEASE.stack to $STACK
+| * Sets stacks.$STACK.releases.$RELEASE.enable to true unless the release has
+|   a "buildpack" property and it's value is no in the stack's install_buildpacks
+|
+| Finally
+|
+| * Merges stacks.$STACK.releases into releases
+| * Validate that all install_stacks and install_buildpacks are defined
+==========================================================================================
+*/}}
+{{- define "_stacks.update" }}
+  {{- /* +---------------------------------------------------------------------------+ */}}
+  {{- /* | Setup the cflinux stack from the configuration in the deployment manifest | */}}
+  {{- /* +---------------------------------------------------------------------------+ */}}
+
+  {{- $_ := include "_config.lookupManifest" (list $ "instance_groups.api.jobs.cloud_controller_ng.properties.cc.stacks") }}
+  {{- $cc_stacks := $.kubecf.retval }}
+
+  {{- if ne (len $cc_stacks) 1 }}
+    {{- fail "cf-deployment defines more than one stack (or none)" }}
```

Comment:
Yes, Diego supports multiple stacks, and so do we. This line is just protecting the code that extracts details for `cflinuxfs3` from `cf-deployment.yml`. It assumes that this manifest only lists a single stack, which has been true so far.

All of this to avoid hard-coding the string `cflinuxfs3` anywhere outside the `config/` directory. (I know there is still one other place).

Do you think this needs to be changed, or can we resolve this as-is?

Reply:
Ah, I somehow missed the cf-deployment-ness (which is… dumb, because the line I chose to comment on specifically mentions cf-deployment). Resolving.